lib/blocs/receipt/receipt_cubit.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:coffeecard/data/repositories/receipt_repository.dart';
+import 'package:coffeecard/models/receipts/receipt.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+part 'receipt_state.dart';
+
+class ReceiptCubit extends Cubit<ReceiptState> {
```

Comment:
While I'm not inherently against using cubits, I personally find bloc more appealing since I think it provides a cleaner separation between the UI and the logic. From a developer perspective, I'm also not sure if I would say they are the same system, since, while the differences are small, you can't implement it the same way, obviously. Do you @jonasanker have further thoughts on it, given the motivation of using it in the first place?

In regard to the function for the `refreashIndicator`, then to get the same functionality using a bloc you could check for the `first` or `firstWhere`, on the stream of states that the bloc has.